Sources/KeyboardShortcuts/NSMenuItem++.swift

```diff
@@ -35,20 +40,46 @@ extension NSMenuItem {
 	- Important: You will have to disable the global keyboard shortcut while the menu is open, as otherwise, the keyboard events will be buffered up and triggered when the menu closes. This is because `NSMenu` puts the thread in tracking-mode, which prevents the keyboard events from being received. You can listen to whether a menu is open by implementing `NSMenuDelegate#menuWillOpen` and `NSMenuDelegate#menuDidClose`. You then use `KeyboardShortcuts.disable` and `KeyboardShortcuts.enable`.
 	*/
 	public func setShortcut(for name: KeyboardShortcuts.Name?) {
-		func clear() {
-			keyEquivalent = ""
-			keyEquivalentModifierMask = []
-		}
-
 		guard let name = name else {
-			clear()
+			clearShortcut()
 			AssociatedKeys.observer[self] = nil
 			return
 		}
 
 		func set() {
-			guard let shortcut = KeyboardShortcuts.Shortcut(name: name) else {
-				clear()
+			let shortcut = KeyboardShortcuts.Shortcut(name: name)
+			setShortcut(shortcut)
+		}
+
+		set()
+
+		// TODO: Use Combine when targeting macOS 10.15.
+		AssociatedKeys.observer[self] = NotificationCenter.default.addObserver(forName: .shortcutByNameDidChange, object: nil, queue: nil) { notification in
+			guard
+				let nameInNotification = notification.userInfo?["name"] as? KeyboardShortcuts.Name,
+				nameInNotification == name
+			else {
+				return
+			}
+
+			set()
+		}
+	}
+
+	/**
+	Add a keyboard shortcut to a `NSMenuItem`.
```

Comment:
Can you mention that the other `setShortcut` method is the recommended one and that this one is useful for dynamic keyboard shortcut where you store the shortcuts yourself. Like, mention your use-case.

Reply:
Yes, of course. Done!